HHS/simpler-grants-gov
# [Issue 2554] clean up fetcher frontend data fetcher pattern

## Summary
Fixes #2554

### Time to review: __15 mins__

## Changes proposed

This change removes the `services/search/searchfetcher` directory, and moves any necessary related functionality into the `/api` classes. For more discussion on reasoning and expectations, refer to the ticket

Note that for ease of code change management this is currently branched off of [dschrashun/2448-v1-search-endpoint
](https://github.com/HHS/simpler-grants-gov/pull/2518) which has not been merged to main yet. If this PR is approved before feature/frontend-opensearch is approved for merging to main, this will merge into freature/frontend-opensearch. This seems fine as the change isn't urgent but could be rebased off main if we want to get it in.

## Context for reviewers

### Test steps

1. _VERIFY_ all search operations work as expected
2. _VERIFY_ opportunity detail pages load as expected

## Additional information
 Follow up todos:

- [ ] refactor opportunity detail page to properly use suspense skeletons
- [ ] figure out a way to properly mock API interactions and allow for page level testing without mocking suspense (try upgrading to next 15 / react 19?)
